src/components/navbar.astro

```diff
@@ -23,11 +23,42 @@
 
         <div class="h-8 w-[1px] bg-gray-400 dark:bg-gray-500"></div>
 
-        <!-- TODO: theme toggle -->
-        <button class="flex h-8 w-8 items-center justify-center">
+        <button class="flex h-8 w-8 items-center justify-center" id="toggle-id">
           <i class="bx bx-moon text-xl"></i>
         </button>
       </div>
     </nav>
   </body>
 </html>
+
+<script is:inline>
+  const TOGGLE_ID = "toggle-id";
+  const INIT = "init";
+  const TOGGLE = "toggle";
+  const DARK = "dark";
+  const LIGHT = "light";
+
+  const button = document.getElementById(TOGGLE_ID);
+  const icon = button?.querySelector("i");
+  if (!button || !icon) throw new Error(`missing "#${TOGGLE_ID}" or its icon`);
```

Comment:
this could fail silently, but there is probably a bigger issue if the required elements don't exist